lib/github.ml

```diff
@@ -260,37 +250,47 @@ let curl_upload_archive ~token ~dry_run curl archive user repo release_id =
 
 let curl_open_pr ~token ~dry_run ~title ~distrib_user ~user ~branch ~body
     ~opam_repo curl =
-  let parse_url resp =
-    (* FIXME this is nuts. *)
-    let url =
-      Re.(
-        compile
-        @@ seq
-             [
-               bol;
-               str {|  "html_url":|};
-               rep space;
-               char '"';
-               group (rep (compl [ char '"' ]));
-             ])
-    in
-    let alread_exists = Re.(compile @@ str "A pull request already exists") in
-    try Ok (`Url Re.(Group.get (exec url resp) 1))
-    with Not_found ->
-      if Re.execp alread_exists resp then Ok `Already_exists
-      else R.error_msgf "Could not find html_url id in response:\n%s." resp
+  let parse_url json =
+    match Yojson.Basic.Util.member "html_url" json with
+    | `String s -> R.ok (`Url s)
+    | `Null -> (
+        match Yojson.Basic.Util.member "message" json with
+        | `String s
+          when String.is_prefix s ~affix:"A pull request already exists" ->
+            R.ok `Already_exists
+        | `Null ->
+            R.error_msgf
+              "Could not find release html url or error message in response:\n\
+               %a."
+              Yojson.Basic.pp json
+        | field ->
+            R.error_msgf
+              "Could not find release html url, and could not parse error \
+               message from header: %S"
+              (Yojson.Basic.to_string field) )
+    | field ->
+        R.error_msgf "Could not parse html url from header: %S"
+          (Yojson.Basic.to_string field)
   in
   let base, repo = opam_repo in
   let uri = strf "https://api.github.com/repos/%s/%s/pulls" base repo in
   let data =
-    strf {|{"title": %S,"base": "master", "body": %S, "head": "%s:%s"}|} title
-      body user branch
+    let open Yojson.Basic in
+    `Assoc
+      [
+        ("title", `String title);
+        ("base", `String "master");
+        ("body", `String body);
+        ("head", `String (strf "%s:%s" user branch));
+      ]
+    |> pretty_to_string
```

Comment:
We can use the more compact `to_string` here, I think it would make more sense!

Reply:
By compact I meant that it produces a compact json string, that wasn't very clear in my first message!

Reply:
done